 *** hibernate-distribution-3.3.1.GA/project/core/src/main/java/org/hibernate/tuple/entity/PojoEntityTuplizer.java       Tue Jan 13 11:19:17 2009
--- ../old/hibernate-distribution-3.3.1.GA/project/core/src/main/java/org/hibernate/tuple/entity/PojoEntityTuplizer.java        Wed Sep 10 12:19:50 2008
***************
*** 28,34 ****
  import java.lang.reflect.Modifier;
  import java.util.HashSet;
  import java.util.Iterator;
- import java.util.LinkedHashSet;
  import java.util.Map;
  import java.util.Set;

--- 28,33 ----
***************
*** 113,119 ****
        protected ProxyFactory buildProxyFactory(PersistentClass persistentClass, Getter idGetter, Setter idSetter) {
                // determine the id getter and setter methods from the proxy interface (if any)
          // determine all interfaces needed by the resulting proxy
!               HashSet proxyInterfaces = new LinkedHashSet();

                Class mappedClass = persistentClass.getMappedClass();
                Class proxyInterface = persistentClass.getProxyInterface();
--- 112,119 ----
        protected ProxyFactory buildProxyFactory(PersistentClass persistentClass, Getter idGetter, Setter idSetter) {
                // determine the id getter and setter methods from the proxy interface (if any)
          // determine all interfaces needed by the resulting proxy
!               HashSet proxyInterfaces = new HashSet();
!               proxyInterfaces.add( HibernateProxy.class );

                Class mappedClass = persistentClass.getMappedClass();
                Class proxyInterface = persistentClass.getProxyInterface();
***************
*** 147,153 ****
                                proxyInterfaces.add( subclassProxy );
                        }
                }
-               proxyInterfaces.add( HibernateProxy.class );

                Iterator properties = persistentClass.getPropertyIterator();
                Class clazz = persistentClass.getMappedClass();
--- 147,152 ----
